library/pod-security-policy/allow-privilege-escalation/template.yaml

```diff
@@ -36,6 +36,35 @@ spec:
                 type: string
   targets:
     - target: admission.k8s.gatekeeper.sh
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: containers
+            expression: 'has(object.spec.containers) ? object.spec.containers : []'
+          - name: initContainers
+            expression: 'has(object.spec.initContainers) ? object.spec.initContainers : []'
+          - name: ephemeralContainers
+            expression: 'has(object.spec.ephemeralContainers) ? object.spec.ephemeralContainers : []'
+          - name: exemptImages
+            expression: |
+              !has(variables.params.exemptImages) ? [] :
+                (variables.containers + variables.initContainers + variables.ephemeralContainers).filter(container,
+                  // we can use the regex substitution because docker images do not allow "." as a valid character
+                  variables.params.exemptImages.exists(exemption, container.image == exemption || (exemption.endsWith("*") && string(container.image).matches("^" + string(variables.params.exemptImages).replace("*", ".*") + "$")))
```

Comment:
I tested with this and got this error `composited variable "badContainers" fails to evaluate: composited variable "exemptImages" fails to evaluate: no such overload: string(list)`
```
 parameters:
    exemptImages: ["*"]
```

Reply:
Gah, should have been `string(exemption)` -- exemptImages is a list

Reply:
fixed